core/src/main/java/org/wildfly/channel/ChannelImpl.java

```diff
@@ -231,9 +242,15 @@ Optional<ResolveLatestVersionResult> resolveLatestVersion(String groupId, String
                     }
                 case MAVEN_LATEST:
                     String latestMetadataVersion = resolver.getMetadataLatestVersion(groupId, artifactId);
+                    if (blocklistedVersions != null && blocklistedVersions.contains(latestMetadataVersion)) {
```

Comment:
I'm not sure how the MAVEN_LATEST and MAVEN_RELEASE should be handled. We can't fall back on previous version of an artifact in those cases, so blocking an artifact version would essentially remove the artifact from a channel.

At the very least, we should document how the blocklist works with different resolution modes - @ivassile could you add description to the spec.adoc please?

Reply:
I made changes to the doc to explain the implementation.

Reply:
sorry, I meant the [spec](https://github.com/wildfly-extras/wildfly-channel/blob/main/doc/spec.adoc#blocking-artifact-versions ) document not the example

Reply:
We didn't change anything in the spec. In the example document we cover how the blocklist works.

Reply:
I changed the spec with the correct behaviour.